dist/init/linux-systemd/README.md

```diff
@@ -6,26 +6,153 @@ if you have any questions. Feel free to prepend to your question
+sudo chown root:root /tmp/caddy-package/caddy
+sudo mv /tmp/caddy-package/caddy /usr/local/bin/
+```
+
+Then install `/etc/systemd/system/caddy.service` and `/etc/tmpfiles.d/caddy.conf`:
+
+```
+curl --fail --silent --show-error --location --remote-name-all \
+  https://raw.githubusercontent.com/mholt/caddy/master/dist/init/linux-systemd/caddy.service
+curl --fail --silent --show-error --location --remote-name-all \
+  https://raw.githubusercontent.com/mholt/caddy/master/dist/init/linux-systemd/caddy.conf
+
+sudo chown root:www-data caddy.service
+sudo chown root:www-data caddy.conf
+
+sudo chmod 0755 caddy.service
+sudo chmod 0755 caddy.conf
+
+sudo mv caddy.service /etc/systemd/system/
+sudo mv caddy.conf /etc/tmpfiles.d/
+```
+
+Have the caddy service start automatically on boot if you like:
+
+```bash
+sudo systemctl daemon-reload
+sudo systemctl start caddy.service
+sudo systemctl enable caddy.service
+```
+
+If caddy doesn't seem to start properly you can view the log data to help figure out what the problem is:
+
+```bash
+journalctl --pager-end --catalog --unit caddy.service
+```
+
+Use `log stdout` and `errors stderr` in your Caddyfile to fully utilize systemd journaling.
+
+If your GNU/Linux distribution does not use *journald* with *systemd* then check any logfiles in `/var/log`.
+
+If you want to follow the latest logs from caddy you can do so like this:
+
+```bash
+journalctl --follow --unit caddy.service
+```
+
+## Directory Structure
+
+We expect that your directory structure for your caddy installation looks like this:
+
+```
+/usr/local/bin/caddy
+/etc/caddy/Caddyfile
+/etc/ssl/caddy/acme
+/etc/ssl/caddy/ocsp
+/etc/tmpfiles.d/caddy.conf
+/srv/www/localhost
+/var/log/caddy
+```
+
+Here's how you can ensure that what we expect matches your reality:
+
+```
+# Optionally create the directories
+sudo mkdir -p /etc/caddy
+sudo mkdir -p /etc/ssl/caddy
+sudo mkdir -p /srv/www/localhost
+sudo mkdir -p /var/log/caddy
+
+# add Caddyfile if it doesn't exist
+if [ ! -f "/etc/caddy/Caddyfile" ]; then
+  sudo bash -c 'echo "http://localhost {
+  root /srv/www/localhost
+}" >> /etc/caddy/Caddyfile'
+fi
+
+# add a default page to serve if it doesn't exist
+if [ ! -f "/srv/www/localhost/index.html" ]; then
+  sudo bash -c 'echo "Welcome to Caddy" >> /srv/www/localhost/index.html'
+fi
+
+# change the permissions
+sudo chown -R root:www-data /usr/local/bin/caddy
+sudo chown -R root:www-data /etc/caddy
+sudo chown -R www-data:www-data /etc/ssl/caddy
+sudo chown -R www-data:www-data /var/log/caddy
+sudo chown -R www-data:www-data /srv/www/localhost
+sudo chown root:www-data /etc/systemd/system/caddy.service
+sudo chown root:www-data /etc/tmpfiles.d/caddy.conf
+
+sudo chmod 0755 /usr/local/bin/caddy
+sudo chmod 0755 /etc/caddy/
+sudo chmod 0644 /etc/caddy/Caddyfile
+sudo chmod 0770 /etc/ssl/caddy
+sudo chmod 0755 /var/log/Caddy
+sudo chmod 0775 /srv/www/
+sudo chown 0644 /etc/systemd/system/caddy.service
+sudo chown 0644 /etc/tmpfiles.d/caddy.conf
```

Comment:
This whole big section is redundant. That's why we have the *tmpfile*:

```bash
mv caddy.conf /etc/tmpfiles.d/
systemd-tmpfiles --create
```

… and done!